avro-fastserde/src/main/java/org/apache/avro/generic/ColdDatumReaderMixIn.java

```diff
@@ -0,0 +1,27 @@
+package org.apache.avro.generic;
+
+import com.linkedin.avro.fastserde.coldstart.ColdPrimitiveFloatList;
+import org.apache.avro.Schema;
+
+
+/**
+ * An interface with default implementation in order to defeat the lack of multiple inheritance.
+ */
+public interface ColdDatumReaderMixIn {
+  default Object newArray(Object old, int size, Schema schema, NewArrayFunction fallBackFunction) {
+    switch (schema.getElementType().getType()) {
+      case FLOAT:
+        if (null == old || !(old instanceof ColdPrimitiveFloatList)) {
+          old = new ColdPrimitiveFloatList(size);
```

Comment:
nit - cam simply return here

Reply:
Sure. I also spotted a bug in this function... fixed it.